File: typescript/src/models/link.ts
// this file is @generated
import { type LinkType, LinkTypeSerializer } from "./linkType";

export interface Link {
  type: LinkType;
  url: string;
}

export const LinkSerializer = {
  _fromJsonObject(object: any): Link {
    return {
      type: LinkTypeSerializer._fromJsonObject(object["type"]),
      url: object["url"],
    };
  },

  _toJsonObject(self: Link): any {
    return {
      type: LinkTypeSerializer._toJsonObject(self.type),
      url: self.url,
    };
  },
};

File: ruby/lib/agentic_commerce_protocol/models/link.rb
# frozen_string_literal: true
# This file is @generated
require "json"

module AgenticCommerceProtocol
  class Link
    attr_accessor :type
    attr_accessor :url

    ALL_FIELD ||= ["type", "url"].freeze
    private_constant :ALL_FIELD

    def initialize(attributes = {})
      unless attributes.is_a?(Hash)
        fail(
          ArgumentError,
          "The input argument (attributes) must be a hash in `AgenticCommerceProtocol::Link` new method"
        )
      end

      attributes.each do |k, v|
        unless ALL_FIELD.include?(k.to_s)
          fail(ArgumentError, "The field #{k} is not part of AgenticCommerceProtocol::Link")
        end

        instance_variable_set("@#{k}", v)
        instance_variable_set("@__#{k}_is_defined", true)
      end
    end

    def self.deserialize(attributes = {})
      attributes = attributes.transform_keys(&:to_s)
      attrs = Hash.new
      attrs["type"] = AgenticCommerceProtocol::LinkType.deserialize(attributes["type"])
      attrs["url"] = attributes["url"]
      new(attrs)
    end

    def serialize
      out = Hash.new
      out["type"] = AgenticCommerceProtocol::serialize_schema_ref(@type) if @type
      out["url"] = AgenticCommerceProtocol::serialize_primitive(@url) if @url
      out
    end

    # Serializes the object to a json string
    # @return String
    def to_json
      JSON.dump(serialize)
    end
  end
end
